src/templates/controller/controller.template.ts

```diff
@@ -1,104 +1,172 @@
-@ApiTags('${serviceName}')
-@Controller('${serviceName}')
-export class ${cServiceName}Controller {
-    constructor(private readonly ${serviceName}Service: ${cServiceName}Service) {}
-    `
-}
+    private controllerFile: string
 
-function getPathWithoutFirstRoute(path: string) {
-    const parts = path.split('/').filter(Boolean);
-  
-    if (parts.length <= 1) {
-      return undefined;
+    private _imports: Record<string, Set<string>> = {
+        ['@nestjs/common']: new Set<string>().add('Controller'),
+        ['@nestjs/swagger']: new Set<string>().add('ApiTags'),
+        ['../models']: new Set<string>(),
+        // the service and dto are imported inside the constructor
     }
-  
-    return parts.slice(1).join('/');
-  }
-
-type addPathProps = ControllerPath & {
-    serviceName: string,
-    imports: any,
-}
 
-const addPath = ({method, path, pathParams, queryParams, serviceName, body, imports, returnType}: addPathProps) => {
-
-    imports.commons.add(_.capitalize(method))
-    if(body) {
-        imports.commons.add('Body')
-        imports.dtos.add(body)
-    }
-    if(returnType) {
-        imports.dtos.add(returnType.includes('[]') ? returnType.slice(0,returnType.length - 2) : returnType)
-    }
-    if(pathParams) {
-        imports.commons.add('Param')
+    get imports() {
+        return this._imports
     }
 
-    if(queryParams) {
-        imports.commons.add('Query')
+    constructor(config: ControllerConfig, rootPath) {
+        this.serviceName = config.serviceName
+        this.rootPath = rootPath
+        this._imports[`./${this.serviceName}.service`] = new Set<string>().add(
+            `${_.capitalize(this.serviceName)}Service`,
+        )
+        this.paths = methodSortFn(config.paths)
+            .map((path) => this.addPath({ ...path }))
+            .join('\n\n')
+
+        this.generateController()
     }
 
-    const methodName = `${methodNames[method.toUpperCase() as keyof typeof methodNames]}${pathParams ? `By${pathParams.map(_.capitalize).join()}` : ''}`
+    private addPath = ({
+        method,
+        path,
+        pathParams,
+        queryParams,
+        body,
+        returnType,
+    }: ControllerPath) => {
+        this._imports['@nestjs/common'].add(_.capitalize(method))
+        if (body) {
+            this._imports['@nestjs/common'].add('Body')
+            this._imports['../models'].add(body)
+        }
+        if (returnType) {
+            this._imports['../models'].add(
+                returnType.includes('[]')
+                    ? returnType.slice(0, returnType.length - 2)
+                    : returnType,
+            )
+        }
+        if (pathParams) {
+            this._imports['@nestjs/common'].add('Param')
+        }
+
+        if (queryParams) {
+            this._imports['@nestjs/common'].add('Query')
+        }
+
+        const methodName = `${methodNames[method.toUpperCase() as keyof typeof methodNames]}${pathParams ? `By${pathParams.map(_.capitalize).join()}` : ''}`
```

Comment:
```suggestion
        const baseMethodName =
            methodNames[method.toUpperCase() as keyof typeof methodNames]
        const byParamSuffix = pathParams
            ? `By${pathParams.map(_.capitalize).join()}`
            : ''
        const methodName = `${baseMethodName}${byParamSuffix}`
```
я бы предложил разбить на несколько строк... цель - избежать путаницы в будущем...